astarte/device/device.py

```diff
@@ -156,6 +157,59 @@ def add_interface(self, interface_definition: dict) -> None:
         """
         self.__introspection.add_interface(interface_definition)
 
+    def add_interface_from_file(self, interface_file: Path):
+        """
+        Adds an interface to the device
+
+        This will add an interface definition to the device. It has to be called before
+        :py:func:`connect`, as it will be used for building the device introspection.
+
+        Parameters
+        ----------
+        interface_file : Path
+            An absolute path to an Astarte interface json file.
+        Raises
+        ------
+        FileNotFoundError
+            If specified file does not exists.
+        TypeError
+            If speficied file is not a .json file.
+        """
+        if not interface_file.is_file():
+            raise FileNotFoundError(f'"{interface_file}" does not exist or is not a file')
+        if interface_file.suffix != ".json":
+            raise TypeError(f'"{interface_file}" is not a json file')
+        with open(interface_file, "r", encoding="utf-8") as interface_fp:
+            self.__introspection.add_interface(json.load(interface_fp))
```

Comment:
`json.load` already raises an error in the file does not contain a serialized JSON and `add_interface_from_dir` already filters on the suffix. I'd replace this with

```suggestion
        try:
            with open(interface_file, "r", encoding="utf-8") as interface_fp:
                self.__introspection.add_interface(json.load(interface_fp))
        except JSONDecodeError:
            raise TypeError(f'"{interface_file}" is not a json file')
```